feat: CRUD Teacher / Migration & Factory "user_id"

LucasDLabo opened the pull request:
- Se realizó el CRUD de Teacher con las correspondientes validaciones y mensajes de error.
- Se añadió la relación con la tabla "city" y "user" para su correcto funcionamiento.
- Se cambio el atributo authorize() dentro de la request a "true" para su correcto funcionamiento.
- Se cambio la columna "user_id" para que pueda ser nullable() para su correcto funcionamiento.
- Se cambio el código de la factory para la tabla "teachers" para que solo se generen campos únicos en "user_id".

LucasDLabo committed:
Fix: Course, User, Career

Fix Users
-Se implementó el Soft Delete
- Se agregó poder ver los usuarios eliminados.

Fix Careers
- Se eliminó las validaciones en el Update.
- Se agregó poder editar la institución.
- Se agregó poder ver las carreras eliminadas.

Fix Course
- Se eliminó las validaciones en el Update.
- Se agregó poder editar la carrera.
- Se agregó poder ver los cursos eliminados.

LucasDLabo committed:
feat: CRUD Teacher / Migration & Factory "user_id"

Se realizó el CRUD de Teacher con las correspondientes validaciones y mensajes de error.
Se añadió la relación con la tabla "city" y "user" para su correcto funcionamiento.
Se cambio el atributo authorize() dentro de la request a "true" para su correcto funcionamiento.
Se cambio la columna "user_id" para que pueda ser nullable() para su correcto funcionamiento.
Se cambio el código de la factory para la tabla "teachers" para que solo se generen campos únicos en "user_id".

MaksiConGit reviewed (changes requested):
- StoreTeacherRequest tiene user_id comentado
- El user_id tiene que seleccionarse por el nombre de usuario (cuentas sin usuarios asignados)
- Crear UpdateTeacherRequest para guardar las validaciones de unique con ignore. Además de poner el resto de validaciones del Store en el Update

LucasDLabo committed:
fix: CRUD Teachers v1

Se cambio el campo 'required' a 'nullable' dentro de StoreTeacherRequest.
Se creó un archivo de validaciones para el update de Teachers "UpdateTeacherRequest".
Se implemento una lista desplegable de los usuarios libres en el Create de profesores.
Se implemento una lista desplegable de los usuarios libres y el usuario actual en el Edit de profesores.

MaksiConGit reviewed (changes requested):
- UpdateTeacherRequest tiene únicamente las validaciones que hacían conflictos con el unique, pero faltan el resto de validaciones que tiene el Store. Por ejemplo, no puedo crear un usuario que de fecha de nacimiento sea mañana, pero sí puedo editar un usuario y poner que su fecha de nacimiento sea mañana.

LucasDLabo committed:
fix: Update Teacher v1.1

Se agregó el resto de validaciones faltantes dentro del UpdateTeacherRequest.

LucasDLabo committed:
fix: Update Teacher v1.2

Se agregó 2 validaciones faltantes dentro de 'dni'. (string y max)

MaksiConGit reviewed (changes requested):
- En el HTML está escrito "career" cuando debería decir "city"
- Si se triggerean las validaciones con algo mal, la ciudad no se guarda con old()

LucasDLabo committed:
fix: Update Teacher v1.3

Se corrigió la label que tenia "career" escrito dentro del Create.
Se agregó la función de mantener el campo previo elegido de city cuando la validación falla.
